java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -174,14 +174,14 @@ public void setInitialCapacity(int valueCount) {
    * @param valueCount desired number of elements in the vector
    * @param density average number of bytes per variable width element
    */
+  @Override
   public void setInitialCapacity(int valueCount, double density) {
-    final long size = (long) (valueCount * density);
-    if (size < 1) {
-      throw new IllegalArgumentException("With the provided density and value count, potential capacity of the data buffer is 0");
-    }
+    long size = Math.max((long)(valueCount * density), 1L);
```

Comment:
I am not 100% sure about this. Why isn't this a illegal argument exception like before?

Reply:
Because it is better to internally set the initial capacity to 1 as opposed to throwing exception.
In our code, we invoke this in a loop dynamically adjusting the density value and stepping down the initial capacity because we are working with fix memory reservation and limits.

So setInitialCapacity() followed by allocateNew() might fail in the second step if not enough memory. So we restart by adjusting the density and stepping down the value count. Throwing an exception doesn't help in any case.

Reply:
Ok. I don't have strong opinion on the behavior for the case `valueCount * density < 1`, I guess what you are saying make sense. 

In that case, can we make that clear in the documentation (maybe at the interface level)? Also, this is consistent with `setInitialCapacity(valueCount)`? i.e Does `setInitialCapacity(0)` also adjust to 1 as well?